substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/target/Target_sun_reflect_annotation_AnnotationParser.java

```diff
@@ -0,0 +1,443 @@
+                }
+                memberValues.put(memberName, value);
+            }
+        }
+        return annotationForMap(annotationClass, memberValues);
+    }
+
+    @Alias
+    public static native Object parseMemberValue(Class<?> memberType,
+                    ByteBuffer buf,
+                    Target_jdk_internal_reflect_ConstantPool constPool,
+                    Class<?> container);
+
+    @Substitute
+    private static Object parseClassValue(ByteBuffer buf,
+                    Target_jdk_internal_reflect_ConstantPool constPool,
+                    Class<?> container) {
+        int classIndex = buf.getShort() & 0xFFFF;
+        try {
+            return constPool.getClassAt(classIndex);
+        } catch (Throwable t) {
+            throw VMError.shouldNotReachHere();
+        }
+    }
+
+    @Substitute
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static Object parseEnumValue(Class<? extends Enum> enumType, ByteBuffer buf,
+                    Target_jdk_internal_reflect_ConstantPool constPool,
+                    Class<?> container) {
+        int typeIndex = buf.getShort() & 0xFFFF;
+        int constNameIndex = buf.getShort() & 0xFFFF;
+        String constName = constPool.getUTF8At(constNameIndex);
+
+        if (enumType != constPool.getClassAt(typeIndex)) {
+            Target_sun_reflect_annotation_AnnotationTypeMismatchExceptionProxy e = new Target_sun_reflect_annotation_AnnotationTypeMismatchExceptionProxy();
+            e.constructor(enumType.getTypeName() + "." + constName);
+            return e;
+        }
+
+        try {
+            return Enum.valueOf(enumType, constName);
+        } catch (IllegalArgumentException e) {
+            return new EnumConstantNotPresentExceptionProxy((Class<? extends Enum<?>>) enumType, constName);
+        }
+    }
+
+    @Substitute
+    private static Object parseConst(int tag,
+                    ByteBuffer buf, Target_jdk_internal_reflect_ConstantPool constPool) {
+        switch (tag) {
+            case 'B':
+                return buf.get();
+            case 'C':
+                return buf.getChar();
+            case 'D':
+                return buf.getDouble();
+            case 'F':
+                return buf.getFloat();
+            case 'I':
+                return buf.getInt();
+            case 'J':
+                return buf.getLong();
+            case 'S':
+                return buf.getShort();
+            case 'Z':
+                byte value = buf.get();
+                assert value == 1 || value == 0;
+                return buf.get() == 1;
+            case 's':
+                return constPool.getUTF8At(buf.getShort());
+            default:
+                throw new AnnotationFormatError(
+                                "Invalid member-value tag in annotation: " + tag);
+        }
+    }
+
+    @Substitute
+    private static Object parseByteArray(int length,
```

Comment:
All of these are needed because we use an `int` for encoding and the JDK uses `short`?

This deserves a comment, both why we use `int` and why we substitute.

Reply:
Added comment